src/Components/Components/src/Routing/RouteEntry.cs

```diff
@@ -25,23 +27,46 @@ public RouteEntry(RouteTemplate template, Type handler, string[] unusedRoutePara
 
         internal void Match(RouteContext context)
         {
-            if (Template.Segments.Length != context.Segments.Length)
+            // If the template contains no parameters, then we want to ensure
+            // that the route and the template are strictly equal to each other.
+            // We only check for non-empty segments in the comparison to avoid
+            // checking for the extraneous "" that exists because we only trim
+            // the starting '/' in a route instead of both the starting and
+            // ending '/'.
+            var hasParameter = Template.Segments.Any(t => t.IsParameter);
+            var lengthNonEmptySegments = context.Segments.Where(t => t != string.Empty).Count();
+            if (!hasParameter && lengthNonEmptySegments != Template.Segments.Length)
```

Comment:
Got some feedback around the impact this check will have on perf for this hot-code path.

I've got some ideas on how this can be spruced that I will commit later to this branch.

Reply:
Turns out the exactness check using the number of matching segments in the code below is essentially doing the same validation as this so this can be removed.